### PR TITLE
Login

### PR DESCRIPTION
**Pull request by Nw84**
We have added a login page with three api routes for creating an acount, logging in and logging out

### CODE REVIEW

File: pages/login/index.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                    <p>Har du inget konto ? Klicka här för att registrera dig</p>
+                    <Button onClick={() => setPageState("registration")}>Registrera Konto</Button>  
+                </div>
+            </div>
+        )
+    } else if (pageState == "registration") {
+        return (
+            <div>
+                <RegistrationForm handleSubmit={handleSubmit} setUser={setUser} setPwd={setPwd} />
+                <div>
```
Comment (emmaambr) [inline pages/login/index.js:83]: "handleSubmit" needs to be changed to "handleRegistration"

File: pages/login/index.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            }),
+            headers: {
+                "Content-Type": "application/json"
+            }
+        });
+        router.push("/login");
+    }
+
+    async function handleRegistration(event) {
+        event.preventDefault();
```
Comment (emmaambr) [inline pages/login/index.js:49]: Got an error message that complained about event.preventDefault. tried removing it and it work fine without.

File: components/login/RegistrationForm.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+function RegistrationForm({ handleRegistration, setUser, setPwd }) {
+  function submitRegistration(event) {
+    event.preventDefault();
```
Comment (emmaambr) [inline components/login/RegistrationForm.js:3]: Got an error message that complained about event.preventDefault. tried removing it and it work fine without.

File: pages/api/users.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+        bcrypt.genSalt(saltRounds, function (err, salt) {
+            bcrypt.hash(req.body.password, salt, async function(err, hash) {
+                const data = {
+                    name: req.body.username,
+                    password: hash
+                }
+            
+                const result = await regUser(data);
+                console.log(result);
```
Comment (emmaambr) [inline pages/api/users.js:18]: Remember to remove console.log() before merging

**Comment by mhema42** [comment]
I got error about that there should be autoComplete for input fields, so I tried to add:
_autoComplete="Ange ditt användarnamn"_ and _autoComplete="Ange ditt lösenord"_, before "placeholder in components/login/LoginForm.js
and _autoComplete="Mitt unika användarnamn"_ and _autoComplete="Mitt säkra lösenord"_ , in components/login/RegistrationForm.js
This fixed that error message, so it may be something to add before merge?

I saw that there is no message _"Felaktigt lösenord"_ or _"Det finns ingen användare med det namnet"_ , if you enter a no existing user or wrong password. I think it looks right in api/login.js, but obviously something is wrong, haven´t found a solution though... :(

**Review by emmaambr — CHANGES_REQUESTED** [review]
View my earlier comments, to see the changes the needs to be done before merging.

File: components/login/RegistrationForm.js (hunk 3 above)
Comment (mhema42) [reply components/login/RegistrationForm.js:3]: Resolved

File: pages/login/index.js (hunk 2 above)
Comment (mhema42) [reply pages/login/index.js:49]: Resolved

File: pages/login/index.js (hunk 1 above)
Comment (mhema42) [reply pages/login/index.js:83]: Resolved

File: pages/api/users.js (hunk 4 above)
Comment (mhema42) [reply pages/api/users.js:18]: Resolved

**Review by simonlj8 — APPROVED** [review]
In chrome addon "React" under "components" i can see username and password when i log in.
Don't now if it's a big problem because it disappears as soon user click elsewhere on the site.
bosse

Our radiobutton "allow cookies" should be necessary to be checked to proceed. Now it dosen't matter if it checked or not.
Some minor fix with styling would do fine. (Username and password input could be little more seperated)
Overall it works just great!

**Pull request merged**